forge-game/src/main/java/forge/game/ability/effects/DiscardEffect.java

```diff
@@ -32,47 +32,58 @@ public class DiscardEffect extends SpellAbilityEffect {
 
     @Override
     protected String getStackDescription(SpellAbility sa) {
-        final String mode = sa.getParam("Mode");
-        final StringBuilder sb = new StringBuilder();
-
-        final Iterable<Player> tgtPlayers = Iterables.filter(getTargetPlayers(sa), PlayerPredicates.canDiscardBy(sa, true));
+        final List<Player> tgtPlayers = Lists.newArrayList(Iterables.filter(getTargetPlayers(sa),
+                PlayerPredicates.canDiscardBy(sa, true)));
```

Comment:
Is there a better way to do this?

Reply:
Hmm, not sure, this is what I would do as well :/

Reply:
getTargetPlayers(sa).filter() is a bit cleaner